### PR TITLE
[QOL-7580] ensure non-PDF files are downloaded rather than displayed inline

### PR DESCRIPTION
**Pull request by ThrawnCA**
HTML and image files may contain embedded code.

Also make the code slightly more compatible with the Keitaro fork.

### CODE REVIEW

File: ckanext/s3filestore/uploader.py [hunk 1]
```diff
@@ -145,7 +145,7 @@ def clear_key(self, filepath):
         except Exception as e:
             raise e
 
-    def get_signed_url_to_key(self, key_path):
+    def get_signed_url_to_key(self, key, extra_params={}):
```
Comment (duttonw) [inline ckanext/s3filestore/uploader.py:148]: what does extra_params do?

Comment (ThrawnCA) [reply ckanext/s3filestore/uploader.py:148]: I put it there to align with the Keitaro fork. It's an optional way to override parameters when calling this function (it does a dict update).

**Comment by duttonw** [comment]
compile errors
https://travis-ci.com/github/qld-gov-au/ckanext-s3filestore/jobs/459309442

  File "/home/travis/virtualenv/python2.7.17/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2449, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/home/travis/build/qld-gov-au/ckanext-s3filestore/ckanext/s3filestore/controller.py", line 122
    if host_name[-1] == '/':
                           ^
SyntaxError: invalid syntax

and may test errros.

**Comment by duttonw** [comment]
please re-base with develop, i've just brought it up to be inline with master.

**Comment by ThrawnCA** [comment]
Superseded by #33

**Pull request closed without merging**